package/run.sh

```diff
@@ -9,12 +9,32 @@ while [ "$STACK_NAME" == "" ]; do
   sleep 1
   STACK_NAME=$(wget -q -O - ${META_URL}/self/stack/name)
 done
+# Get etcd service certificates
+UUID=$(curl -s http://rancher-metadata/2015-12-19/stacks/Kubernetes/services/etcd/uuid)
+ACTION=$(curl -s -u $CATTLE_ACCESS_KEY:$CATTLE_SECRET_KEY "$CATTLE_URL/services?uuid=$UUID" | jq -r '.data[0].actions.certificate')
+
+if [ -n "$ACTION" ]; then
+    mkdir -p /etc/etcd/ssl
+    cd /etc/etcd/ssl
+    curl -s -u $CATTLE_ACCESS_KEY:$CATTLE_SECRET_KEY -X POST $ACTION > certs.zip
+    unzip -o certs.zip
+    cd $OLDPWD
+
+fi
+
+export ETCD_CA_FILE="/etc/etcd/ssl/ca.pem"
+export ETCD_KEY_FILE="/etc/etcd/ssl/key.pem"
+export ETCDCTL_CA_FILE="/etc/etcd/ssl/ca.pem"
+export ETCD_CERT_FILE="/etc/etcd/ssl/cert.pem"
+export ETCDCTL_KEY_FILE="/etc/etcd/ssl/key.pem"
+export ETCDCTL_CERT_FILE="/etc/etcd/ssl/cert.pem"
 
 SCALE=$(giddyup service scale etcd)
 
 while [ ! "$(echo $IP | grep -E '^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$')" ]; do
     sleep 1
     IP=$(wget -q -O - ${META_URL}/self/container/primary_ip)
```

Comment:
lets use rancher metadata ip address instead of fqdn. Make the ip configurable, with the default (check other infra services like healthcheck for example)